vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/AndroidTestRunListener.kt

```diff
@@ -3,23 +3,25 @@ package com.malinskiy.marathon.android.executor.listeners
 import com.malinskiy.marathon.android.model.TestIdentifier
 
 interface AndroidTestRunListener {
-    suspend fun beforeTestRun() {}
+    suspend fun beforeTestRun()
 
-    suspend fun testRunStarted(runName: String, testCount: Int) {}
+    suspend fun testRunStarted(runName: String, testCount: Int)
 
-    suspend fun testStarted(test: TestIdentifier) {}
+    suspend fun testStarted(test: TestIdentifier)
 
-    suspend fun testFailed(test: TestIdentifier, trace: String) {}
+    suspend fun testFailed(test: TestIdentifier, trace: String)
 
-    suspend fun testAssumptionFailure(test: TestIdentifier, trace: String) {}
+    suspend fun testAssumptionFailure(test: TestIdentifier, trace: String)
 
-    suspend fun testIgnored(test: TestIdentifier) {}
+    suspend fun testIgnored(test: TestIdentifier)
 
-    suspend fun testEnded(test: TestIdentifier, testMetrics: Map<String, String>) {}
+    suspend fun testEnded(test: TestIdentifier, testMetrics: Map<String, String>)
 
-    suspend fun testRunFailed(errorMessage: String) {}
+    suspend fun testRunFailed(errorMessage: String)
 
-    suspend fun testRunStopped(elapsedTime: Long) {}
+    suspend fun testRunStopped(elapsedTime: Long)
 
-    suspend fun testRunEnded(elapsedTime: Long, runMetrics: Map<String, String>) {}
+    suspend fun testRunStoppedFailFast(elapsedTime: Long)
```

Comment:
Why do we need an additional method for this particular case?

Reply:
I make this for the cleanliness of my changes. So, should I use `testRunStopped`?

Reply:
I think we shouldn't add more methods to an already complex lifecycle unless required. Otherwise we need to change all our screenshot/log/filesync listeners